src/main/java/com/syncstream_dev/syncstream/model/user/User.java

```diff
@@ -0,0 +1,62 @@
+package com.syncstream_dev.syncstream.model.user;
+
+import java.time.LocalDateTime;
+
+
+public class User {
+    private UserInfo userInfo;
+    private String uid;     // User id
+    private String username;
+    private int messagesSent;
+    private int points;
+
+    // Register User Constructor
+    public User(String uid, String username, String token, String email, String password, String avatarUrl) {
+        // Todo: Database integration
+        this.userInfo = new UserInfo(token, email, password, LocalDateTime.now(), LocalDateTime.now(), avatarUrl);
+        this.uid = uid;
+        this.username = username;
+        this.messagesSent = 0;
+        this.points = 0;
+    }
+
+    // Login User Constructor
+    public User(String username, String password) {
```

Comment:
`password` is preferably a `char` or `byte` array.

Reminder for future: Encryption - hashing and salts.

Reply:
char array

Reply:
88cc85045b3b0d788200ab3e4557935c049ef00f addresses this